### DISCUSSION TITLE
mini.indentscope: how to change the color?

### DISCUSSION THREAD

**Discussion by dprophete** [nvim-mini/mini.nvim#725] [General]
Wondering if there is a way to customize the indentation line color. Couldn't find anything in the doc. Thanks.

**Comment by dprophete** [comment]
Ok, I am replying to my own question here... It might help others. Just use something like:
```
hi MiniIndentscopeSymbol guifg=#888888
```

**Reply by JettIsOnTheNet** [reply]
Having looked at your comment, and it being the top search result, hopefully this will help other people who want to keep their theme values:

IMHO a different solution that works with your theme and falls back automatically would be to use vim.g.terminal_color* vals.

So, inside your mini-indentscope.lua:

```lua
local symbol_color = vim.g.terminal_color_2 or '#FFFFFF'
vim.api.nvim_create_autocmd('ColorScheme', {
    pattern = "*",
    callback = function()
        vim.api.nvim_set_hl(0, 'MiniIndentscopeSymbol', { fg = symbol_color, nocombine = true })
})
```

Theme colors are:
```lua
g.terminal_color_0 = colors.black
g.terminal_color_1 = colors.red
g.terminal_color_2 = colors.green
g.terminal_color_3 = colors.yellow
g.terminal_color_4 = colors.purple
g.terminal_color_5 = colors.pink
g.terminal_color_6 = colors.cyan
g.terminal_color_7 = colors.white
g.terminal_color_8 = colors.selection
g.terminal_color_9 = colors.bright_red
g.terminal_color_10 = colors.bright_green
g.terminal_color_11 = colors.bright_yellow
g.terminal_color_12 = colors.bright_blue
g.terminal_color_13 = colors.bright_magenta
g.terminal_color_14 = colors.bright_cyan
g.terminal_color_15 = colors.bright_white
g.terminal_color_background = colors.bg
g.terminal_color_foreground = colors.fg
```

Here is an example of how to integrate it into the config:

```lua
return {
  {
    'echasnovski/mini.indentscope',
    version = false,
    opts = {
      draw = {
        delay = 100,
        animation = 
      },
     symbol = '▏',
     options = { try_as_border = true },
    },
    init = function()
      vim.api.nvim_create_autocmd('ColorScheme', {
        pattern = '*',
        callback = function()
          -- terminal_color_2 (green) or fallback to white
          local symbol_color = vim.g.terminal_color_2 or '#FFFFFF'
          vim.api.nvim_set_hl(0, 'MiniIndentscopeSymbol', {
            fg = symbol_color,
            nocombine = true,
          })
        end,
      })
    end,
  },
}
```

**Reply by echasnovski** [reply]
> Having looked at your comment, and it being the top search result, hopefully this will help other people who want to keep their theme values:

To be fair, default value [is already aligned with color scheme's colors](https://github.com/echasnovski/mini.nvim/blob/c404fc5717866ff738aadeeb1f54ab8ed270ae1e/lua/mini/indentscope.lua#L701), as it links to built-in group. However, adjusting its color in a way that works with any color scheme indeed requires an autocommand for a `ColorScheme` event.

One more thing to add here is that I'd still suggest to use linking to some built-in highlight group, as it is a more robust solution. In this case, `vim.g.terminal_color_2` usually means green color. This is also the case for `DiagnosticOk` highlight group. So I'd use something like `vim.api.nvim_set_hl(0, 'MiniIndentscopeSymbol', { link = 'DiagnosticOk' })`.

**Reply by JettIsOnTheNet** [reply]
Sure, and yes, but some may prefer a different value than the existing alignment. I have built a number of my own themes, so I tend to use the direct values. Using a highlight group totally makes sense though. Thanks for the good plugin, and reply.

edit: I just realize you're in Kharkiv. Stay safe friend. Слава Україні!
